core/src/main/java/com/scalar/db/common/AbstractTwoPhaseCommitTransactionManager.java

```diff
@@ -12,34 +11,35 @@
 import com.scalar.db.api.TwoPhaseCommitTransactionManager;
 import com.scalar.db.api.Update;
 import com.scalar.db.api.Upsert;
-import com.scalar.db.common.error.CoreError;
 import com.scalar.db.config.DatabaseConfig;
-import com.scalar.db.exception.transaction.AbortException;
-import com.scalar.db.exception.transaction.CommitException;
+import com.scalar.db.exception.transaction.CommitConflictException;
+import com.scalar.db.exception.transaction.CrudConflictException;
 import com.scalar.db.exception.transaction.CrudException;
-import com.scalar.db.exception.transaction.PreparationException;
+import com.scalar.db.exception.transaction.PreparationConflictException;
 import com.scalar.db.exception.transaction.RollbackException;
 import com.scalar.db.exception.transaction.TransactionException;
+import com.scalar.db.exception.transaction.TransactionNotFoundException;
 import com.scalar.db.exception.transaction.UnknownTransactionStatusException;
-import com.scalar.db.exception.transaction.ValidationException;
+import com.scalar.db.exception.transaction.ValidationConflictException;
+import com.scalar.db.util.ScalarDbUtils;
+import com.scalar.db.util.ThrowableFunction;
 import java.util.List;
 import java.util.Optional;
-import java.util.concurrent.CopyOnWriteArrayList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public abstract class AbstractTwoPhaseCommitTransactionManager
-    implements TwoPhaseCommitTransactionManager, TwoPhaseCommitTransactionDecoratorAddable {
+    implements TwoPhaseCommitTransactionManager {
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(AbstractDistributedTransactionManager.class);
```

Comment:
```suggestion
      LoggerFactory.getLogger(AbstractTwoPhaseCommitTransactionManager.class);
```

Reply:
Good catch! Fixed in 194e18dae02328c1901b4d9b5254f80044b0adaa. Thanks.